go/adapter/honeycomb/adapter.go

```diff
@@ -0,0 +1,134 @@
+package honeycomb
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+
+	observe "github.com/dylibso/observe-sdk/go"
+	otel "github.com/dylibso/observe-sdk/go/adapter/otel_formatter"
+	trace "go.opentelemetry.io/proto/otlp/trace/v1"
+	proto "google.golang.org/protobuf/proto"
+)
+
+type HoneycombConfig struct {
+	ApiKey             string
+	Dataset            string
+	EmitTracesInterval uint32
+	TraceBatchMax      uint32
+	Host               string
+}
+
+type HoneycombAdapter struct {
+	*observe.AdapterBase
+	Config *HoneycombConfig
+}
+
+func NewHoneycombAdapter(config *HoneycombConfig) *HoneycombAdapter {
+	base := observe.NewAdapterBase(1, 0)
+	adapter := &HoneycombAdapter{
+		AdapterBase: &base,
+		Config:      config,
+	}
+
+	adapter.AdapterBase.SetFlusher(adapter)
+
+	return adapter
+}
+
+func (h *HoneycombAdapter) Start() {
+	h.AdapterBase.Start(h)
+}
+
+func (h *HoneycombAdapter) HandleTraceEvent(te observe.TraceEvent) {
+	h.AdapterBase.HandleTraceEvent(te)
+}
+
+func (h *HoneycombAdapter) Flush(evts []observe.TraceEvent) error {
+	for _, te := range evts {
+		traceId := te.TelemetryId.ToHex16()
+
+		var allSpans []*trace.Span
+		for _, e := range te.Events {
+			switch event := e.(type) {
+			case observe.CallEvent: // TODO: consider renaming to FunctionCall for consistency across Rust & JS
+				spans := h.makeCallSpans(event, nil, traceId)
+				if len(spans) > 0 {
+					allSpans = append(allSpans, spans...)
+				}
+			case observe.MemoryGrowEvent:
+				log.Println("MemoryGrowEvent should be attached to a span")
+			case observe.CustomEvent:
+				log.Println("Otel adapter does not respect custom events")
+			}
+		}
+
+		if len(allSpans) <= 1 {
+			log.Println("No spans built for honeycomb")
+			return nil
```

Comment:
just checking, this _must_ have more than 1 span?

Reply:
Hmm good catch, this was copy pasta'd from another adapter.  @bhelx do you know of any reason this shouldn't be `len(allSpans) == 0` ?

Reply:
no i assume this is a mistake

Reply:
please do driveby change on source as well